src/keybinds.cpp

```diff
@@ -50,6 +51,19 @@ void check_keybinds(struct overlay_params& params, uint32_t vendorID){
       }
    }
 
+   if (elapsedPresetToggle >= keyPressDelay &&
+       keys_are_pressed(params.toggle_preset)) {
+     toggle_preset_press = now;
+     size_t size = params.preset.size();
```

Comment:
this seems over-engineered. Consider this instead:
Add default to the presets switch
```
      default:
         params->preset = 0;
         presets(0, params);
```
This way way we don't need extra variables like `current_preset` and we don't need to check if preset is valid.
Just use `params->preset` and `++` it. If that makes sense?

Reply:
That would only work if the user doesn't specify a preset variable in the config file.  If it's in the config file, `presets()` returns before ever getting to the switch.  The idea behind this change is to allow the user to specify which presets to cycle through.  For example, if I set presets to `preset=3,5,2`, 3 is the default preset, and when I hit the toggle presets keybind, it cycles to 5, then 2, and then back to 3

Reply:
how about
```current_preset = params.preset[++current_preset % size];```?